ha/core/system_health/system_health.py

```diff
@@ -402,15 +403,17 @@ def process_event(self, healthevent: HealthEvent):
                     # If health is already stored and its a node_health, check further
                     stored_genration_id = current_health_dict["events"][0]["specific_info"]["generation_id"]
                     incoming_generation_id = healthevent.specific_info["generation_id"]
-                    incoming_health_status = current_health_dict["events"][0]["status"]
+                    stored_health_status = current_health_dict["events"][0]["status"]
+                    incoming_health_status = healthevent.event_type
                     pod_restart_val = current_health_dict["events"][0]["specific_info"]["pod_restart"]
                     # Update the current health value itself.
                     latest_health = EntityHealth.read(current_health)
                     if stored_genration_id != incoming_generation_id:
-                        if incoming_health_status == status:
-                            # If the generation id matches and stored node health matches
-                            # with incoming node health, means online event received first
-                            # instead of failed event in delete scenario
+                        if incoming_health_status == HEALTH_EVENTS.ONLINE.value:
+                            # In delete scenario, online event comes first, followed by failed event.
```

Comment:
@mariyappanp Is this section of code specific to node events? If not, will this change affect other components?

Reply:
generation_id is specific to node event. It is not expected to be populated for other event types. 
However its better to have the resource_type check also added here so that disk and other events do not go inside this check,

Reply:
I have added `component_type == "node"` check. @ArchanaLimaye @indrajitzagade 
Please review.

Reply:
@ArchanaLimaye The node related events coming from hare may not have generation_id in the event, right?
Let's discuss this change before merge. @mariyappanp

Reply:
As discussed, I have updated the code. @indrajitzagade @ArchanaLimaye